lib/storage/src/content_manager/consensus_state.rs

```diff
@@ -614,6 +630,14 @@ impl Persistent {
         self.save()
     }
 
+    pub fn last_applied_entry(&self) -> Option<u64> {
+        if self.snapshot_was_applied {
+            Some(self.state.hard_state.commit)
```

Comment:
Why can't we always take the `Some(self.state.hard_state.commit)`?

Reply:
So there are two cases.

_Case 1_
Raft tells us to commit N messages (it has already changed the commit index), but this might take long time so we save our progress in UnappliedEntries in case the node might be shutdown during the process.

_Case 2_
Raft tells us to apply a snapshot -  in that case we apply it at once and don't have any progress.